web/src/engine/main/src/keymanEngine.ts

```diff
@@ -310,12 +310,22 @@ export default class KeymanEngine<
   }
 
   public getDebugInfo(): Record<string, any> {
+    const activeKbd = this.contextManager.activeKeyboard;
+
     const report = {
-      configReport: this.config.debugReport()
-      // oskType / oskReport?
-      // - mode
-      // - dimensions
-      // other possible entries?
+      configReport: this.config.debugReport(),
+      keyboard: {
+        id: unprefixed(activeKbd.metadata.id),
+        langId: activeKbd?.metadata.langId || '',
+        version: activeKbd?.keyboard.version
+      },
+      model: {
+        id: this.core.activeModel?.id || ''
+      },
+      osk: {
+        banner: this.osk?.bannerController.activeType,
+        layer: this.osk?.vkbd?.layerId || ''
+      }
```

Comment:
```suggestion
      configReport: this.config?.debugReport(),
      keyboard: {
        id: unprefixed(activeKbd?.metadata?.id ?? ''),
        langId: activeKbd?.metadata?.langId || '',
        version: activeKbd?.keyboard?.version ?? ''
      },
      model: {
        id: this.core?.activeModel?.id || ''
      },
      osk: {
        banner: this.osk?.bannerController?.activeType ?? '',
        layer: this.osk?.vkbd?.layerId || ''
      }
```

Given this is an error reporting scenario, where we are already in undefined behaviour territory, I wonder if we should be doing `?.` for all our object paths, to avoid possibility of cascade errors in unexpected scenarios -- e.g. metadata object has failed to construct, etc, etc.